### ISSUE TITLE
Dynamic imports

### ISSUE THREAD

**Issue by JeffHerb** [standard-things/esm#71]
Sorry if this is the wrong place, but this module has made my ES6 development so much better, except for one thing; dynamic imports. Im trying do something like:

```
    for (let name of this.moduleList) {

        import (fullName)
            .then(() => {
                console.log("Done!");
            })
            .catch((e) => {
                console.log("Import error");
            });

    }
```

Is this possible yet?

**Comment by hville** [comment]
In the meantime there is [node-es-module-loader](https://www.npmjs.com/package/node-es-module-loader) based on `babel` and [node-dynamic-import](https://www.npmjs.com/package/node-dynamic-import) based on `rollup`

**Comment by jdalton** [comment]
@hville while I appreciate the drive, know the time and place. Unprompted drive by promotion of other projects aren't super great. Esp when we cover this use case.

**Comment by jdalton** [comment]
Hi @JeffHerb!

Yes, it should be. I have an optimization in place in the parser which might be to blame if it's not hitting. Do you have a repro-repo that I can look at?

**Comment by JeffHerb** [comment]
I dont have a repo for this project just yet, but does this help?

```
class Plugins {

    constructor() {
        this.builtin = ['test']
    }

    load() {

        if (this.builtin.length) {

            for (let name of this.builtin) {

                import (name)
                    .then(() => {
                        console.log("Done!");
                    })
                    .catch((e) => {
                        console.log("Import error");
                    });

            }

        }

}

export default Plugins;
```

**Comment by jdalton** [comment]
Yep, [that was it](https://github.com/standard-things/esm/commit/7f14800c3e05f22c43a09d27ae1092eacad68ce3). Expect an update within the week. I have some refactoring to do before so it may take a day or so.

**Comment by jdalton** [comment]
Fixed by v0.6.0. Please note I've [changed some usage instructions](https://github.com/standard-things/esm#getting-started).
